lib/ide/ide/src/view/ui_component.rs

```diff
@@ -0,0 +1,18 @@
+//! This module contains the base definition of UiComponent.
+
+use nalgebra::Vector2;
+
+/// A trait defining common interfaces of UI components.
+pub trait UiComponent {
```

Comment:
Not sure but weren't our naming convetion to keep all acronym's letter capitalized? (`UIComponent` in this case)

Reply:
@mwu-tow I think we covenanted that. We had a similar case with `HtmlRenderer` (now `Css3dRenderer`)